php/libraries/LorisForm.class.inc

```diff
@@ -1180,7 +1180,7 @@ class LorisForm
 
                 if (isset($el['required'])
                     && $el['required'] === true
-                    && empty($value)
+                    && (is_null($value) || (empty($value) && $value !=0))
```

Comment:
In fact the whole thing could probably be: 
```php
if (!(empty($el['required']) || empty($value)) {
...
```
aka. "If neither variable is empty," which I think is more readable and does what you want.